deployment/ccip/changeset/deployer_group.go

```diff
@@ -74,32 +105,10 @@ func (d *DeployerGroup) GetDeployer(chain uint64) (*bind.TransactOpts, error) {
 		Context:    txOpts.Context,
 		AccessList: txOpts.AccessList,
 		NoSend:     true,
+		// Set the gas limit to avoid simulating the call before appending to transactions.
+		// Some calls depend on each other, so we don't want to simulate until we actually start executing.
+		GasLimit: 1_000_000,
```

Comment:
Just a thought, would it make sense to allow simulating the changeset before executing on the enact part? Maybe not something for this PR but that could add value overall. I am also a bit sad that we have to remove the simulation from here, that was a nice feature

Reply:
I've adjusted it a bit to address your concerns: https://github.com/smartcontractkit/chainlink/pull/15877/commits/ed01ffef43b864be426d151e837ce0ecabd9d7f9

If deployer key is used, simulations now happen when transactions are appended. I've split my token admin registry changeset into 3 different proposals to pass simulations. The deployer group now takes opCount offsets into account to enable enacting multiple proposals.

I was wondering though, do you think simulations should happen if MCMS key is used? Right now they are not because gasLimit is hard-set in SimTransactOpts.

Reply:
Two backlog tickets for deployer group @carte7000 

https://smartcontract-it.atlassian.net/browse/CCIP-5018
https://smartcontract-it.atlassian.net/browse/CCIP-5019

Reply:
I think there is value of having simulation even with MCMS, but that could be something we work out after if for now it's blocking this PR